dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -132,6 +132,10 @@ public boolean isValidWorkflow(String content) {
                 importFile.setType(SourceFile.FileType.DOCKSTORE_WDL);
                 imports.put(importFile.getPath(), importFile);
             }
+
+            for (SourceFile sf : imports.values()) {
+                imports.putAll(processImports(sf.getContent(), version, sourceCodeRepoInterface));
```

Comment:
I don't know if this is actually possible, and it certainly wouldn't be a good WDL practice, but if a imported b, and b imported a, couldn't that lead to an infinite loop, which might eventually bring down the Dockstore?

I see you have a catch, but I wonder if that would be too late to recover from.

Reply:
If the catch is the IOException, that won't work well. 
It certainly would tie up a lot of CPU, it might even blow out temp space in the hotfix branches. 

Since we have the map of imports processed so far in `imports`, just check to make sure you don't call processImports on a file that's already in it.

Reply:
I tested it and there is an infinite loop. I've updated the code to deal with this and it properly pulls the files, however it seems like the WDL4S parsing code for the tool table and DAG will also need to be fixed.

Reply:
Ok, tool table and DAG can wait for develop. 
Imports is hotfix (as it is here)